### PR TITLE
Produce errors instead of producing bad autoplot() plots

### LINKED ISSUE #755: Avoid warning from `geom_line()` bubbling up in `autoplot()`?
Do we want to try to avoid that warning from `geom_line()`?

``` r
library(tidymodels)
library(censored)
#> Loading required package: survival
library(finetune)

set.seed(1)
sim_dat <- prodlim::SimSurv(500) %>%
  mutate(event_time = Surv(time, event)) %>%
  select(event_time, X1, X2)

sim_rs <- vfold_cv(sim_dat)

time_points <- c(10, 1, 5, 15)

mod_spec <-
  decision_tree(tree_depth = tune(), min_n = 4) %>%
  set_engine("partykit") %>%
  set_mode("censored regression")

sint_mtrc <- metric_set(brier_survival_integrated)

grid <- tibble(tree_depth = c(1, 2, 10))
rctrl <- control_race(save_pred = TRUE, verbose_elim = FALSE, verbose = FALSE)

set.seed(2193)
aov_integrated_res <-
  mod_spec %>%
  tune_race_anova(
    event_time ~ X1 + X2,
    resamples = sim_rs,
    grid = grid,
    metrics = sint_mtrc,
    eval_time = time_points,
    control = rctrl
  )

autoplot(aov_integrated_res, eval_time = 10) 
#> `geom_line()`: Each group consists of only one observation.
#> ℹ Do you need to adjust the group aesthetic?
```

<sup>Created on 2023-07-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
To close #755 

``` r
library(tidymodels)
library(censored)
#> Loading required package: survival
library(finetune)

set.seed(1)
sim_dat <- prodlim::SimSurv(500) %>%
  mutate(event_time = Surv(time, event)) %>%
  select(event_time, X1, X2)

sim_rs <- vfold_cv(sim_dat)

time_points <- c(10, 1, 5, 15)

mod_spec <-
  decision_tree(tree_depth = tune(), min_n = 4) %>%
  set_engine("partykit") %>%
  set_mode("censored regression")

sint_mtrc <- metric_set(brier_survival_integrated, brier_survival)

grid <- tibble(tree_depth = c(1, 2, 10))
rctrl <- control_race(save_pred = TRUE, verbose_elim = FALSE, verbose = FALSE)

set.seed(2193)

aov_integrated_res <-
  mod_spec %>%
  tune_race_anova(
    event_time ~ X1 + X2,
    resamples = sim_rs,
    grid = grid,
    metrics = sint_mtrc,
    eval_time = time_points,
    control = rctrl
  )
#> Warning: Evaluation times are only required when dynmanic or integrated metrics are
#> selected as the primary metric (and will be ignored).

autoplot(aov_integrated_res, eval_time = 10) 
#> Error in `plot_regular_grid()`:
#> ! Only one observation per metric was present. Unable to create
#>   meaningful plot.
```

### CODE REVIEW

**Comment by topepo** [comment]
> #> Warning: Evaluation times are only required when dynmanic or integrated metrics are
> #> selected as the primary metric (and will be ignored).

Ug 🙈

File: R/plots.R [hunk 1]
```diff
@@ -510,6 +510,14 @@ plot_regular_grid <- function(x, metric = NULL, eval_time = NULL, ...) {
   dat <- paste_param_by(dat)
   dat <- dat %>% dplyr::filter(!is.na(mean))
   dat <- filter_plot_eval_time(dat, eval_time)
+
+  if (all(vctrs::vec_count(dat$.metric)$count == 1)) {
+    cli::cli_abort(
```
Comment (topepo) [inline R/plots.R:515]: Can we avoid 

> Error in `plot_regular_grid()`:

by passing the calling environment through?

Comment (EmilHvitfeldt) [reply R/plots.R:515]: sure thing, i was on the fence, whether to do it now or wait for the {cli} push

Comment (EmilHvitfeldt) [reply R/plots.R:518]: done in [`01c6ded` (#801)](https://github.com/tidymodels/tune/pull/801/commits/01c6ded9b770bd7299dda048f7cc1d7229f0885a)

**Pull request merged**